ably/realtime_channel.go

```diff
@@ -167,9 +167,40 @@ func (c *RealtimeChannel) mayAttach(result, checkActive bool) (Result, error) {
 			return nopResult, nil
 		}
 	}
-	if !c.client.Connection.lockIsActive() {
-		return nil, c.state.set(StateChanFailed, errAttach)
+
+	switch c.client.Connection.State() {
+	// RTL4b
+	case StateConnClosed,
+		StateConnClosing,
+		StateConnFailed:
+		return nil, newError(80000, errAttach)
+
+	// RTL4i
+	case StateConnInitialized,
```

Comment:
>If the connection state is CONNECTING or DISCONNECTED, do the operation once the connection state is CONNECTED

I don't see spec mentioning  `StateConnInitialized` why is it here?

Reply:
True, [RTL4b](https://docs.ably.io/client-lib-development-guide/features/#RTL4b) says that would be an error. Fixed at 6fb73b8. (Tests to ensure compliance will follow in the 1.2 branch.)